WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SettingsUsernameChangerFragment.kt

```diff
@@ -0,0 +1,193 @@
+
+import android.app.ProgressDialog
+import android.os.Bundle
+import android.text.Editable
+import android.text.SpannableStringBuilder
+import android.text.Spanned
+import android.text.TextWatcher
+import android.view.ContextThemeWrapper
+import android.view.WindowManager
+import android.widget.EditText
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import androidx.core.text.HtmlCompat
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.R
+import org.wordpress.android.R.style
+import org.wordpress.android.analytics.AnalyticsTracker
+import org.wordpress.android.analytics.AnalyticsTracker.Stat.ACCOUNT_SETTINGS_CHANGE_USERNAME_FAILED
+import org.wordpress.android.analytics.AnalyticsTracker.Stat.ACCOUNT_SETTINGS_CHANGE_USERNAME_SUCCEEDED
+import org.wordpress.android.analytics.AnalyticsTracker.Stat.ACCOUNT_SETTINGS_CHANGE_USERNAME_SUGGESTIONS_FAILED
+import org.wordpress.android.fluxc.generated.AccountActionBuilder
+import org.wordpress.android.fluxc.store.AccountStore.AccountUsernameActionType.KEEP_OLD_SITE_AND_ADDRESS
+import org.wordpress.android.fluxc.store.AccountStore.OnUsernameChanged
+import org.wordpress.android.fluxc.store.AccountStore.PushUsernamePayload
+import org.wordpress.android.ui.FullScreenDialogFragment.FullScreenDialogController
+import org.wordpress.android.util.AccessibilityUtils
+import org.wordpress.android.util.ActivityUtils
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T
+import org.wordpress.android.widgets.WPDialogSnackbar
+
+/**
+ * Allows the user to change their username from the Account Settings screen.
+ */
+class SettingsUsernameChangerFragment : BaseUsernameChangerFullScreenDialogFragment() {
+    private lateinit var dialogController: FullScreenDialogController
+    private var progressDialog: ProgressDialog? = null
+
+    override fun getSuggestionsFailedStat() = ACCOUNT_SETTINGS_CHANGE_USERNAME_SUGGESTIONS_FAILED
+    override fun canHeaderTextLiveUpdate() = false
+    override fun getHeaderText(username: String?, display: String?): Spanned = HtmlCompat.fromHtml(
+            String.format(
+                    getString(R.string.settings_username_changer_header),
+                    "<b>",
+                    username,
+                    "</b>"
+            ), HtmlCompat.FROM_HTML_MODE_LEGACY
+    )
+
+    override fun onViewCreated(controller: FullScreenDialogController) {
+        super.onViewCreated(controller)
+        dialogController = controller
+
+        dialogController.setActionEnabled(false)
+    }
+
+    /**
+     * The Save Action will only be enabled when a new username has been selected.
+     */
+    override fun onUsernameSelected(username: String?) {
+        super.onUsernameSelected(username)
+        dialogController.setActionEnabled(mUsernameSelectedIndex != 0)
+    }
+
+    override fun onConfirmClicked(controller: FullScreenDialogController?): Boolean {
```

Comment:
I always feel weird when I need to override a method without calling super. What do you think about making the method abstract in the parent?
Tbh I'm not a fan of inheritance and I haven't used it in the view layer for quite a while (it has good usages, but it's very easy to misuse). Please tell me if there is something I'm missing or if you think it's not a good idea / best practice. :) I'm looking forward hearing your thoughts about this.

Reply:
Yep, it definitely make sense. Keeping each PR small makes reviews much easier. I usually do one of the following depending on the specific case:
- I create a new ticket with a list of checkboxes for each task I'm planning to take care of later
- I add TODOs to do code which I fix in a next PR
- I mention my plan in the description of the PR explaining that I didn't make the change to keep the scope small
- I make the change in the same PR, but I add a GitHub comment explaining unclear changes. Eg. that I've just moved a method into a different class.

All of these can be helpful for the reviewer and clarifies your intent.

Feel free to make the change either in this PR or in a new PR. It's up to you.